### ISSUE TITLE
Investigate root cause of overwritten MCap polyfill on Safari

### ISSUE THREAD

**Issue by joeyparrish** [google/shaka-player#3696]
In #3530, the MCap polyfill is being overwritten on Safari.  It's not clear what is causing this.  @joeyparrish could not reproduce the issue, but @theodab could.

PR #3668 gave us a workaround, but we still haven't addressed the root cause.  It would be preferable to find the root cause of the issue, revert the PR, and merge a fix that addresses the root cause.

**Comment by joeyparrish** [comment]
I'm able to reproduce the original issue on Safari 15.1.  Previously, I was unable to reproduce it in Safari 14.1.

**Comment by joeyparrish** [comment]
The issue is timing-related.  With breakpoints enabled in the polyfill installation, the problem vanishes.

**Comment by joeyparrish** [comment]
Delaying the polyfill by 5 seconds seems to work around the issue.  With that, content shared with us privately via #3530 gets past the original 6010 error (and instead fails a license request).

**Comment by joeyparrish** [comment]
I was trying to determine if/when/how the property for decodingInfo was being overwritten, and I may have stumbled onto a good clue.

I first changed the polyfill to:
  1. Set decodingInfo
  2. Use [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) to make the property non-writeable
  3. Check 5 seconds later if the value was still what I set it to

After 5 seconds, the value had been reset every time (presumably by the platform).  But I couldn't be sure if it was decodingInfo or the mediaCapabilities parent of decodingInfo which was being reset.

So then I changed the polyfill to:
  1. Set decodingInfo
  2. Make a local copy of the value of mediaCapabilities
  3. Use [`Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) as before
  4. Check both the values of decodingInfo and mediaCapabilities after 5 seconds

And magically... both were the same now!  And consistently when reloading the page to try again.

So here's my theory.  It may be that navigator.mediaCapabilities is a getter which returns a new MediaCapabilities object on-demand, which is later garbage-collected when not referenced.  Creating a local copy that is used later in a closure seems to keep our modifications in the polyfill from being undone.

I'll experiment with this more and see if this could be a viable fix.
